maven-core/src/main/java/org/apache/maven/lifecycle/internal/MojoExecutor.java

```diff
@@ -282,6 +308,32 @@ private Lock getProjectLock( MavenSession session )
         }
     }
 
+    static class OwnerReentrantLock extends ReentrantLock
+    {
+        @Override
+        public Thread getOwner()
+        {
+            return super.getOwner();
+        }
+    }
+
+    static class OwnerReentrantReadWriteLock extends ReentrantReadWriteLock
+    {
+        @Override
+        public Thread getOwner()
+        {
+            return super.getOwner();
+        }
+    }
+
+    private static void warn( String msg )
+    {
+        for ( String s : MessageHelper.formatWarning( msg ) )
+        {
+            LOGGER.warn( s );
+        }
```

Comment:
In a concurrent build may it happen that interleaving threads might break this message as a whole?

Reply:
Yes, but I don't think slf4j provide any api to _lock_ the appender.  In addition, any build-in new line will break the formatting box.  But in a concurrent build, you definitely want to use _mvnd_ which brings back things in order :-)

Reply:
Well, look into /maven-embedder/src/main/java/org/apache/maven/cli/transfer/ConsoleMavenTransferListener.java, I do synchronize there to make console output project. Since the logger is static final you can use it as monitor, no?